core/chains/evm/logpoller/log_poller.go

```diff
@@ -57,8 +57,31 @@ type LogPoller interface {
 	IndexedLogsTopicRange(eventSig common.Hash, address common.Address, topicIndex int, topicValueMin common.Hash, topicValueMax common.Hash, confs int, qopts ...pg.QOpt) ([]Log, error)
 	IndexedLogsWithSigsExcluding(address common.Address, eventSigA, eventSigB common.Hash, topicIndex int, fromBlock, toBlock int64, confs int, qopts ...pg.QOpt) ([]Log, error)
 	LogsDataWordRange(eventSig common.Hash, address common.Address, wordIndex int, wordValueMin, wordValueMax common.Hash, confs int, qopts ...pg.QOpt) ([]Log, error)
-	LogsDataWordGreaterThan(eventSig common.Hash, address common.Address, wordIndex int, wordValueMin common.Hash, confs int, qopts ...pg.QOpt) ([]Log, error)
-	LogsUntilBlockHashDataWordGreaterThan(eventSig common.Hash, address common.Address, wordIndex int, wordValueMin common.Hash, untilBlockHash common.Hash, qopts ...pg.QOpt) ([]Log, error)
+	LogsDataWordGreaterThan(eventSig common.Hash, address common.Address, wordIndex int, wordValueMin common.Hash, confs BlockConfsOpt, qopts ...pg.QOpt) ([]Log, error)
+}
+
+type BlockConfsOpt struct {
+	// useFinalityDepth is used to indicate that only finalized block will be returned, can't be used together with confs
+	// When LogPoller's useFinalityTag is set to true, actual finality based on the chain state will be used.
+	// Otherwise, LogPoller will pick up the finality depth from the config.
+	useFinalityDepth bool
```

Comment:
Simpler to just call this `finalized`. In the probabilistic finality chain case, the best we can do is have a specific number of confs anyways

Reply:
An alternative design might be worth considering, the geth style magic number approach:
```
type Confirmations int

const Finalized = Confirmations(-1)

func Select(.., confs Confirmations) {
	if confs == Finalized {
		confs = lp.getFinalityDepth()
	}
	...
}

lp.Select(..., lp.Finalized)
```
Feels slightly more ergonomic for clients and you can drop the Validate